earth2mip/inference_ensemble.py

```diff
@@ -258,7 +260,20 @@ def perturb(x, rank, batch_id, device):
             )
         if rank == 0 and batch_id == 0:  # first ens-member is deterministic
             noise[0, :, :, :, :] = 0
+
+        center = torch.tensor(
+            [channel_means[channel] for channel in model.in_channel_names],
+            device=x.device,
+        )
+        scale = torch.tensor(
+            [channel_stds[channel] for channel in model.in_channel_names],
+            device=x.device,
+        )
+        center_reshaped = center[:, None, None]
+        scale_reshaped = scale[:, None, None]
+        x = (x - center_reshaped) / scale_reshaped
         x += noise
+        x = (x * scale_reshaped) + center_reshaped
```

Comment:
use this instead so noise with amplitude 0 is allowed, also saves memory.

```suggestion
        x += noise * scale[:, None, None]
```

Reply:
you can also delete channel means above.